src/plugins/intel_cpu/src/emitters/snippets/x64/jit_loop_emitters.cpp

```diff
@@ -12,110 +13,238 @@ using namespace dnnl::impl::cpu::x64;
 namespace ov {
 namespace intel_cpu {
 
-inline static void transform_idxs_to_regs(const std::vector<size_t>& idxs, std::vector<Reg64>& regs) {
+inline static void transform_idxs_to_regs(const std::vector<size_t>& idxs, std::vector<Xbyak::Reg64>& regs) {
     regs.resize(idxs.size());
-    std::transform(idxs.begin(), idxs.end(), regs.begin(), [](size_t idx){return Reg64(static_cast<int>(idx));});
+    std::transform(idxs.begin(), idxs.end(), regs.begin(), [](size_t idx){ return Xbyak::Reg64(static_cast<int>(idx)); });
 }
 
-jit_loop_begin_emitter::jit_loop_begin_emitter(jit_generator* h, cpu_isa_t isa, const ov::snippets::lowered::ExpressionPtr& expr)
-    : jit_emitter(h, isa) {
-    loop_begin = ov::as_type_ptr<snippets::op::LoopBegin>(expr->get_node());
-    OPENVINO_ASSERT(loop_begin != nullptr, "jit_loop_begin_emitter invoked with invalid op argument");
-    const auto loop_end = get_loop_end(expr);
-    work_amount = loop_end->get_work_amount();
-    evaluate_once = loop_end->get_evaluate_once();
+/* ================== jit_loop_begin_emitter ====================== */
+
+jit_loop_begin_emitter::jit_loop_begin_emitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl::cpu::x64::cpu_isa_t isa,
+                                               const ov::snippets::lowered::ExpressionPtr& expr)
+    : jit_emitter(h, isa), loop_begin_label{new Xbyak::Label()} {
     in_out_type_ = emitter_in_out_map::gpr_to_gpr;
 }
 
-std::shared_ptr<snippets::op::LoopEnd> jit_loop_begin_emitter::get_loop_end(const ov::snippets::lowered::ExpressionPtr& expr) {
+std::shared_ptr<ov::snippets::op::LoopEnd> jit_loop_begin_emitter::get_loop_end(const ov::snippets::lowered::ExpressionPtr& expr) {
     OPENVINO_ASSERT(expr->get_output_port_connectors().size() == 1, "jit_loop_begin_emitter has invalid LoopBegin expression configuration");
     const auto& consumers = expr->get_output_port_connector(0)->get_consumers();
     OPENVINO_ASSERT(consumers.size() == 1, "jit_loop_begin_emitter has invalid LoopBegin expression configuration");
-    const auto loop_end = ov::as_type_ptr<snippets::op::LoopEnd>(consumers.cbegin()->get_expr()->get_node());
-    OPENVINO_ASSERT(loop_end != nullptr, "jit_loop_begin_emitter has invalid LoopBegin expression configuration");
-    return loop_end;
+    const auto loop_end_expr = consumers.cbegin()->get_expr();
+    OPENVINO_ASSERT(ov::is_type<snippets::op::LoopEnd>(loop_end_expr->get_node()), "jit_loop_begin_emitter has invalid LoopBegin expression configuration");
+    return ov::as_type_ptr<snippets::op::LoopEnd>(loop_end_expr->get_node());
```

Comment:
It this implementation better than the old one? It looks like the new one does exactly the same as the old one, but performs both `is_type` and `as_type_ptr` casts (which is clearly redundant). What is the reason behind this change?